src/torchmetrics/functional/image/psnrb.py

```diff
@@ -0,0 +1,218 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Tuple, Union
+
+import torch
+from torch import Tensor, tensor
+from typing_extensions import Literal
+
+from torchmetrics.utilities import rank_zero_warn, reduce
+
+
+def _compute_bef(
+    self, target: Tensor, dim: Optional[Union[int, Tuple[int, ...]]] = None, block_size=8
+) -> Tuple[Tensor, Tensor]:
+
+    if dim == 3:
+        height, width, channels = target.Size
+    elif dim == 2:
+        height, width = target.Size
+        channels = 1
+    else:
+        raise ValueError("Not a 1-channel/3-channel grayscale image")
+
+    if channels > 1:
+        raise ValueError("Not for color images")
+
+    h = torch.tensor(range(0, width - 1))
+    h_b = torch.tensor(range(block_size - 1, width - 1, block_size))
+    h_bc = torch.tensor(list(set(h).symmetric_difference(h_b)))
+
+    v = torch.tensor(range(0, height - 1))
+    v_b = torch.tensor(range(block_size - 1, height - 1, block_size))
+    v_bc = torch.tensor(list(set(v).symmetric_difference(v_b)))
+
+    d_b = 0
+    d_bc = 0
+
+    # h_b for loop
+    for i in list(h_b):
+        diff = target[:, i] - target[:, i + 1]
+        d_b += torch.sum(torch.square(diff))
+
+    # h_bc for loop
+    for i in list(h_bc):
+        diff = target[:, i] - target[:, i + 1]
+        d_bc += torch.sum(torch.square(diff))
+
+    # v_b for loop
+    for j in list(v_b):
+        diff = target[j, :] - target[j + 1, :]
+        d_b += torch.sum(torch.square(diff))
+
+    # V_bc for loop
+    for j in list(v_bc):
+        diff = target[j, :] - target[j + 1, :]
+        d_bc += torch.sum(tensor.square(diff))
```

Comment:
I think this should be able to be vectorized in some way, possible using `torch.gather`

Reply:
ok willl try